tests/templates/test_integration.py

```diff
@@ -176,558 +138,347 @@
+             'theta': qml.init.interferometer_theta_uniform(n_wires=1),
+             'wires': range(1)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_normal(n_wires=1),
+             'varphi': qml.init.interferometer_varphi_normal(n_wires=1),
+             'theta': qml.init.interferometer_theta_normal(n_wires=1),
+             'wires': range(1)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_uniform(n_wires=3),
+             'varphi': qml.init.interferometer_varphi_uniform(n_wires=3),
+             'theta': qml.init.interferometer_theta_uniform(n_wires=3),
+             'wires': range(3)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_normal(n_wires=3),
+             'varphi': qml.init.interferometer_varphi_normal(n_wires=3),
+             'theta': qml.init.interferometer_theta_normal(n_wires=3),
+             'wires': range(3)})
+           ]
+
+
+class TestIntegrationQnode:
+    """Tests the integration of templates into qnodes when differentiable arguments are passed as
+    primary or auxiliary arguments to the qnode, using different interfaces."""
+
+    @pytest.mark.parametrize("template, diffable, nondiffable", QUBIT_DIFFABLE_NONDIFFABLE)
     @pytest.mark.parametrize("interface, to_var", INTERFACES)
-    def test_integration_qubit_diffable(self, template1, diffable1, nondiffable1,
-                                        template2, diffable2, nondiffable2,
-                                        interface, to_var):
-        """Tests integration of qubit templates passing differentiable arguments as positional arguments to qnode."""
-        #TODO: rewrite test to avoid quadratic growth of test cases with the number of templates
+    def test_qubit_qnode_primary_args(self, template, diffable, nondiffable, interface, to_var):
+        """Tests integration of qubit templates with other operations, passing differentiable arguments
+        as primary arguments to qnode."""
 
         # Extract keys and items
-        keys_diffable1 = [*diffable1]
-        diffable1 = list(diffable1.values())
-        keys_diffable2 = [*diffable2]
-        diffable2 = list(diffable2.values())
-
-        # Combine diffable inputs to allow passing with *
-        diffable = diffable1 + diffable2
+        keys_diffable = [*diffable]
+        diffable = list(diffable.values())
 
         # Turn into correct format
         diffable = [to_var(i) for i in diffable]
 
         # Generate qnode
         dev = qml.device('default.qubit', wires=2)
-        circuit = qnode_qubit_args(dev, interface, template1, template2, len(diffable1))
+
+        @qml.qnode(dev, interface=interface)
+        def circuit(*diffable, keys_diffable=None, nondiffable=None):
+            # Turn diffables back into dictionary
+            dict = {key: item for key, item in zip(keys_diffable, diffable)}
```

Comment:
Is overwriting `dict` problematic?

Reply:
I think this is why the tests are failing.

Reply:
That is a very good point - it seemed to work but with the last changes it might have stopped. Anyways, better to be safe!